cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -813,7 +813,8 @@ protected boolean checkAnyRunning(Predicate<Id.Program> predicate, ProgramType..
     for (ProgramType type : types) {
       for (Map.Entry<RunId, ProgramRuntimeService.RuntimeInfo> entry :  runtimeService.list(type).entrySet()) {
         ProgramController.State programState = entry.getValue().getController().getState();
-        if (programState == ProgramController.State.STOPPED || programState == ProgramController.State.ERROR) {
+        if (programState == ProgramController.State.COMPLETED || programState == ProgramController.State.ERROR
```

Comment:
Probably better to add a method `ProgramController.State.isDone()` to tell if the given state means the program is done so that you can have `if (programState.isDone())` here.